llvm/utils/TableGen/DAGISelMatcherEmitter.cpp

```diff
@@ -125,6 +128,39 @@ class MatcherTableEmitter {
          [](const auto &A, const auto &B) { return A.second > B.second; });
     for (const auto &PatternPredicate : PatternPredicateList)
       PatternPredicates.push_back(PatternPredicate.first);
+
+    // Sort Predicates by usage.
+    // Merge predicates with same code.
+    for (const auto &Usage : PredicateUsage) {
+      TreePattern *TP = Usage.first;
+      TreePredicateFn Pred(TP);
+      NodePredicatesByCodeToRun[Pred.getCodeToRunOnSDNode()].push_back(TP);
```

Comment:
Iterating over `PredicateUsage` and creating `StringMap NodePredicatesByCodeToRun` can cause non-determinism.